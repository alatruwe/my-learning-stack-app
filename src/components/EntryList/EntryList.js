import React from "react";
import entries from "../../entries.js";
import Entry from "../Entry/Entry.js";

export default class ItemList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      items: [],
      touched: false,
    };
  }
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  // API call here to get data

  render() {
    return (
      <section className="wrapper">
        <ul className="entry-list">
          {entries.map((entry) => (
            <li key={entry.id}>
              <Entry
                date={entry.date}
                mood={entry.mood}
                id={entry.id}
                tech={entry.tech}
                learningNotes={entry.learningNotes}
                strugglingNotes={entry.strugglingNotes}
              />
            </li>
          ))}
        </ul>
      </section>
    );
  }
}
