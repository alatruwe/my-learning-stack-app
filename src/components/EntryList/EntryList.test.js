import React from "react";
import ReactDOM from "react-dom";
import EntryList from "./EntryList";
import renderer from "react-test-renderer";

describe(`EntryList component`, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<EntryList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<EntryList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
