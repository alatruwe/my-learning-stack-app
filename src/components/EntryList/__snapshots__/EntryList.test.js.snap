// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryList component renders the UI as expected 1`] = `
<section
  className="wrapper dashboard-wrapper"
>
  <h2
    className="card-title"
  >
    Dashboard
  </h2>
  <ul
    className="entry-list"
  />
</section>
`;
